### PR TITLE
Upgrade golangci-lint to v1.31.0

### LINKED ISSUE #1951: Upgrade Golang Lint to 1.30.0
## Describe the proposal
Upgrade Golang Lint to 1.30.0

### PR DESCRIPTION
**Pull request by bvwells**
# Description

Upgrade golangci-lint to v1.31.0

## Issue reference

This will close #1951 

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Unit tests passing
* [x] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Comment by bvwells** [comment]
Work in progress.... Need to push further changes to fix other linting issues.

**Comment by bvwells** [comment]
Should pass golangci-lint now...

File: pkg/grpc/api_test.go [hunk 1]
```diff
@@ -204,10 +203,10 @@ func startDaprAPIServer(port int, testAPIServer *api, token string) *grpc_go.Ser
...
 }
 
-func createTestClient(port int) *grpc_go.ClientConn {
-	var opts []grpc_go.DialOption
-	opts = append(opts, grpc_go.WithInsecure())
-	conn, err := grpc_go.Dial(fmt.Sprintf("localhost:%d", port), opts...)
+func createTestClient(port int) *grpc.ClientConn {
+	var opts []grpc.DialOption
+	opts = append(opts, grpc.WithInsecure())
+	conn, err := grpc.Dial(fmt.Sprintf("localhost:%d", port), opts...)
```
Comment (tcnghia) [inline pkg/grpc/api_test.go:209]:
```suggestion
	conn, err := grpc.Dial(fmt.Sprintf("localhost:%d", port), grpc.WithInsecure())
```

**Review by tcnghia — CHANGES_REQUESTED** [review]
Thanks a lot for making this change

File: .golangci.yml [hunk 2]
```diff
@@ -237,3 +238,8 @@ linters:
     - testpackage
     - goerr113
     - nestif
+    - nlreturn
+    - exhaustive
+    - noctx
+    - gci
+    - gofumpt
```
[the remark this answers is not in the document]
Comment (tcnghia) [reply .golangci.yml:245]: Nevermind, I saw https://github.com/dapr/dapr/issues/1951#issuecomment-702924258 now .

**Comment by tcnghia** [comment]
@bvwells I created a PR to have the linter annotated the PR directly https://github.com/dapr/dapr/pull/2186. Feel free to incorporate into this PR, or we can check it / rebase independently. Thanks

File: pkg/grpc/api_test.go (hunk 1 above)
Comment (bvwells) [reply pkg/grpc/api_test.go]: Bit outside the fix, but sure, updated.

File: pkg/actors/actors.go [hunk 3]
```diff
@@ -698,7 +697,8 @@ func (a *actorsRuntime) evaluateReminders() {
 			go func(wg *sync.WaitGroup, reminders []Reminder) {
 				defer wg.Done()
 
-				for _, r := range reminders {
+				for i := range reminders {
+					r := reminders[i]
```
[the remark this answers is not in the document]
Comment (bvwells) [reply pkg/actors/actors.go]: Added comment on all instances of this.

**Review by tcnghia — APPROVED** [review]

**Review by youngbupark — APPROVED** [review]
LGTM. Thanks for your contribution!

**Comment by bvwells** [comment]
Thanks all for the review and merge!

**Pull request merged**